kerchunk/intro.qmd

```diff
@@ -0,0 +1,69 @@
+
+Learn more about kerchunk at [kerchunk.readthedocs.io](https://kerchunk.readthedocs.io/en/latest/).
+
+## How to kerchunk 
+
+[`kerchunk`](https://fsspec.github.io/kerchunk) is a python library you can use to create a reference file from any of the file formats it supports. At time of writing, Kerchunk can create reference files for NetCDF4/HDF5, GRIB2, FITS, TIFF and NetCDF3. Check the [File format backends](https://fsspec.github.io/kerchunk/reference.html#file-format-backends) section of the kerchunk documentation for updates to supported formats.
+
+The reference file is used by the `fsspec.ReferenceFileSystem` to read data from local or remote storage.
+
+Here's a simple example:
+
+```python
+import fsspec
+import json
+from kerchunk.hdf import SingleHdf5ToZarr
+
+local_file = 'some_data.nc'
+out_file = 'some_references.json'
+
+# Instantiate the local file system with fsspec to save kerchunk's reference data as json.
+fs = fsspec.filesystem('')
+in_file = fs.open(local_file)
+
+# The inline threshold adjusts the size below which binary blocks are included directly in the output.
+# A higher inline threshold can result in a larger json file but faster loading time.
+h5chunks = SingleHdf5ToZarr(in_file, local_file, inline_threshold=300)
+with fs.open(out_file, 'wb') as f:
+    f.write(json.dumps(h5chunks.translate()).encode())
+```
+
+::: {.callout-note}
+The powerful [`fsspec` library](https://filesystem-spec.readthedocs.io/en/latest/) provides a uniform file system interface to many different storage backends and protocols. In addition to abstracting existing protocols, its [`ReferenceFileSystem`](https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.implementations.reference.ReferenceFileSystem) class lets you view byte ranges of some other file as a file system. Kerchunk generates these `ReferenceFileSystem` objects.
+:::
+
+
+Kerchunk generates a "reference set" which is a set of references to data or URLs under a key value store that matches the Zarr spec. For example, a simple reference file for a NetCDF file might look like:
+
+```json
+{
+  ".zgroup": "{\n    \"zarr_format\": 2\n}",
+  ".zattrs": "{\n    \"Conventions\": \"UGRID-0.9.0\n\"}",
+  "x/.zattrs": "{\n    \"_ARRAY_DIMENSIONS\": [\n        \"node\"\n ...",
+  "x/.zarray": "{\n    \"chunks\": [\n        9228245\n    ],\n    \"compressor\": null,\n    \"dtype\": \"<f8\",\n  ...",
+  "x/0": ["s3://bucket/path/file.nc", 294094376, 73825960]
+}
+```
+The `["s3://bucket/path/file.nc", 294094376, 73825960]` is the key part, which says that to load the first chunk in the `x` dimension, the Zarr reader needs to fetch a byte range starting at `294094376` with a length of `73825960` bytes. This allows for efficient cloud-native data access without using the standard NetCDF driver.
+
+## Why Kerchunk?
```

Comment:
This should probably be before the reference file example so people get why they care about the rest.